src/task4feedback/simulator/schedulers/parla/state.py

```diff
@@ -1091,3 +1091,236 @@ def completion_stats(self, task: SimulatedTask):
 
                 data.stats.move_time += task.duration
                 data.stats.move_count += 1
+
+
+@SchedulerOptions.register_state("rl")
+@dataclass(slots=True)
+class RLState(SystemState):
```

Comment:
I think this might be from my misunderstanding on the decorator?
If I inherit "ParlaState", is this class registered as "parla" or "rl"?
If the decorator doesn't care about its base class, and this class is registered as only "rl", then I will modify that.

Reply:
It'll work for this (architecture/state) decorator as long as it's in the same class hierarchy since it's building a string to class type dictionary. 

You're right that some of the other decorators like the `register_graph` would care about the type of the arguments in the signature and might have issues as written with this kind of inheritance. But the state/arch should be fine.